### ISSUE TITLE
Options>data - add missing information about data declaration on the columns level

### ISSUE THREAD

**Issue by AMBudnik** [handsontable/docs#174]
### Description
URL: https://handsontable.com/docs/8.1.0/Options.html#data
Reference: handsontable/support#254

### Tasks
* add examples on settings up data on `columns` level

```js
columns: [
  { data: 'id' }
]
```

```js
columns: [{
      data: 1
}]
```

* add types string and number for data ( `{data: 'id'} / {data: 1 }` ) - reference https://github.com/handsontable/handsontable/blob/7bfc83f14f2844e1b12960f6edc22cec2ba758f5/handsontable.d.ts#L1594 definition file. Define what types of data cannot be used.

* define the difference between the data on the instance level and the data on the `columns` level

**Comment by AMBudnik** [comment]
We should also add an example with a data reference by a dot notation.

Examples

array data
https://jsfiddle.net/AMBudnik/te0wfnho/

 ```js
data: [
   {id: 1, name: ['Anna', 'Lorena']},
   {id: 2, name: ['Tom', 'Anders']}
],
rowHeaders: true,
colHeaders: true,
columns: [
   { data: 'name.1'}
]
```

object data https://jsfiddle.net/AMBudnik/te0wfnho/1/

```js
data: [
  {id: 1, name: {first: 'Anna', last: 'Lorena'}},
  {id: 2, name: {first: 'Tom', last: 'Anders'}}
],
rowHeaders: true,
colHeaders: true,
columns: [
  { data: 'name.last'}
]
```

**Comment by scarletfog** [comment]
@AMBudnik I am not quite sure we should include this information directly at: https://handsontable.com/docs/8.1.0/Options.html#data

It was already a bit misleading, the description did not match the types:
![image](https://user-images.githubusercontent.com/20072336/96127490-b536ab00-0ef4-11eb-92e8-8457f19a3401.png)

Maybe we should elaborate more on this topic inside https://handsontable.com/docs/8.1.0/Options.html#columns and then add a link to it in https://handsontable.com/docs/8.1.0/Options.html#data with a brief description? Something like: "`data` can also be of a function type when declared on the `column` level, see more at....".

What do you think?

**Comment by AMBudnik** [comment]
Yup, `columns` can be a better place. We want to keep things clean and avoid ambiguity.

**Comment by warpech** [comment]
It sould be in both places. As a developer, I would search https://handsontable.com/docs/8.2.0/Options.html#data if I wanted to read about the `data` property.

**Comment by AMBudnik** [comment]
All the above ideas are still valid for the new documentation.
